tests/api/controllers/gps_speed/gps_speed.c

```diff
@@ -6,45 +6,31 @@
 
 #define TIME_STEP 32
 
+void check_speed(WbDeviceTag gps, float v, float vx, float vy, float vz, char *end_msg) {
+  double speed = wb_gps_get_speed(gps);
+  ts_assert_double_equal(speed, v, "The speed value measured by the GPS should be %f and not %lf %s.", v, speed, end_msg);
```

Comment:
Double values comparison could depend on the system.
Instead of checking if two values are equals, you should use the `ts_assert_double_in_delta` and `ts_assert_vec3_in_delta` functions instead.